pkg/controllers/jobset_controller.go

```diff
@@ -142,28 +143,54 @@ func (r *JobSetReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(r)
 }
 
-func (r *JobSetReconciler) constructJobFromTemplate(js *jobset.JobSet, rjob *jobset.ReplicatedJob) (*batchv1.Job, error) {
-	job := &batchv1.Job{
-		ObjectMeta: metav1.ObjectMeta{
-			Labels:      make(map[string]string),
-			Annotations: make(map[string]string),
-			Name:        genJobName(js, rjob),
-			Namespace:   js.Namespace,
-		},
-		Spec: *rjob.Template.Spec.DeepCopy(),
-	}
+func (r *JobSetReconciler) constructJobsFromTemplate(js *jobset.JobSet, rjob *jobset.ReplicatedJob) ([]*batchv1.Job, error) {
+	var jobs []*batchv1.Job
 
-	// If enableDNSHostnames is set, update job spec to set subdomain as
-	// job name (a headless service with same name as job will be created later).
-	if rjob.Network.EnableDNSHostnames != nil && *rjob.Network.EnableDNSHostnames {
-		job.Spec.Template.Spec.Subdomain = job.Name
+	// Defaulting and validation.
+	// TODO: Do defaulting and validation in webhook instead of here (https://github.com/kubernetes-sigs/jobset/issues/6)
+	replicas := 1
+	if rjob.Replicas != nil && *rjob.Replicas > 0 {
+		replicas = *rjob.Replicas
+	}
+	labels := rjob.Template.Labels
```

Comment:
We should always copy the labels and annotations to avoid updating the template itself. 

Also, In case we do the job creates in parallel in the future, we should do the copy inside the loop below so that the maps are shared between all instances.

Reply:
oh, are not* shared

Reply:
Makes sense, done.